### PR TITLE
API-878: Added password helper on edit app page

### PR DESCRIPTION
**Pull request by pchasle**
**Description (for Contributor and Core Developer)**

Added password helper on edit app page

image

image

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Apps/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml [hunk 1]
```diff
@@ -63,6 +63,15 @@ akeneo_apps:
         credentials:
             title: Credentials
             helper: Lorem ipsum dolor sit amet, consectetur adipiscing elit
+            password_placeholder: ••••••••
+            password_helper:
+                message: If you forgot your password, you can generate a new one here.
+                link: More info…
+                link_url: https://www.akeneo.com/
```
Comment (fitn) [inline src/Akeneo/Apps/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml]: I don't feel it's necessary to translate this

File: src/Akeneo/Apps/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml [hunk 2]
```diff
@@ -63,6 +63,15 @@ akeneo_apps:
...
+            clear_password_helper:
+                message: Make sure to save this password somewhere as we won’t show it to you twice.
+                link: More info…
+                link_url: https://www.akeneo.com/
```
Comment (fitn) [inline src/Akeneo/Apps/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml]: neither this one

File: src/Akeneo/Apps/front/src/application/apps/components/AppCredentials.tsx [hunk 3]
```diff
@@ -38,11 +37,33 @@ export const AppCredentials: FC<Props> = ({code, credentials: credentials}: Prop
                     {credentials.username}
                 </CopiableCredential>
                 {credentials.password ? (
-                    <CopiableCredential label={translate('akeneo_apps.app.password')}>
+                    <CopiableCredential
+                        label={translate('akeneo_apps.app.password')}
+                        helper={
+                            <InlineHelper warning>
+                                <Translate id='akeneo_apps.edit_app.credentials.clear_password_helper.message' />{' '}
+                                <a href={translate('akeneo_apps.edit_app.credentials.clear_password_helper.link_url')}>
```
Comment (fitn) [inline src/Akeneo/Apps/front/src/application/apps/components/AppCredentials.tsx]: Directly use the url here, you don't need to translate it

Comment (pchasle) [reply src/Akeneo/Apps/front/src/application/apps/components/AppCredentials.tsx]: Usually you may want to point to a different URL for each locale (translated doc/article) and that's why the URL is part of the translations.
Even if our doc is not translated (yet?) it's, I think, a good habit.

File: src/Akeneo/Apps/front/src/application/apps/reducers/apps-reducer.ts [hunk 4]
```diff
@@ -1,24 +1,41 @@
...
+import {Actions, APP_WITH_CREDENTIALS_FETCHED, APP_UPDATED, APPS_FETCHED} from '../actions/apps-actions';
 
 export interface State {
     [code: string]: App & AppCredentials;
 }
 
 export const reducer: Reducer<State, Actions> = (state, action) => {
     switch (action.type) {
+        case APPS_FETCHED:
+            console.log(action);
```
Comment (willy-ahva) [inline src/Akeneo/Apps/front/src/application/apps/reducers/apps-reducer.ts]: RM

**Pull request merged**